pkg/ottl/contexts/ottlspanlink/span_links.go

```diff
@@ -0,0 +1,213 @@
+	return TransformContext{
+		spanLink: spanLink,
+		span:                 span,
+		instrumentationScope: instrumentationScope,
+		resource:             resource,
+		cache:                pcommon.NewMap(),
+	}
+}
+
+func (tCtx TransformContext) GetSpanLink() ptrace.SpanLink {
+	return tCtx.spanLink
+}
+
+func (tCtx TransformContext) GetSpan() ptrace.Span {
+	return tCtx.span
+}
+
+func (tCtx TransformContext) GetInstrumentationScope() pcommon.InstrumentationScope {
+	return tCtx.instrumentationScope
+}
+
+func (tCtx TransformContext) GetResource() pcommon.Resource {
+	return tCtx.resource
+}
+
+func (tCtx TransformContext) getCache() pcommon.Map {
+	return tCtx.cache
+}
+
+func NewParser(functions map[string]ottl.Factory[TransformContext], telemetrySettings component.TelemetrySettings, options ...Option) (ottl.Parser[TransformContext], error) {
+	p, err := ottl.NewParser[TransformContext](
+		functions,
+		parsePath,
+		telemetrySettings,
+		ottl.WithEnumParser[TransformContext](parseEnum),
+	)
+	if err != nil {
+		return ottl.Parser[TransformContext]{}, err
+	}
+	for _, opt := range options {
+		opt(&p)
+	}
+	return p, nil
+}
+
+type StatementsOption func(*ottl.Statements[TransformContext])
+
+func WithErrorMode(errorMode ottl.ErrorMode) StatementsOption {
+	return func(s *ottl.Statements[TransformContext]) {
+		ottl.WithErrorMode[TransformContext](errorMode)(s)
+	}
+}
+
+func NewStatements(statements []*ottl.Statement[TransformContext], telemetrySettings component.TelemetrySettings, options ...StatementsOption) ottl.Statements[TransformContext] {
+	s := ottl.NewStatements(statements, telemetrySettings)
+	for _, op := range options {
+		op(&s)
+	}
+	return s
+}
+
+func parseEnum(val *ottl.EnumSymbol) (*ottl.Enum, error) {
+	if val != nil {
+		if enum, ok := internal.SpanSymbolTable[*val]; ok {
+			return &enum, nil
+		}
+		return nil, fmt.Errorf("enum symbol, %s, not found", *val)
+	}
+	return nil, fmt.Errorf("enum symbol not provided")
+}
+
+func parsePath(val *ottl.Path) (ottl.GetSetter[TransformContext], error) {
+	if val != nil && len(val.Fields) > 0 {
+		return newPathGetSetter(val.Fields)
+	}
+	return nil, fmt.Errorf("bad path %v", val)
+}
+
+func newPathGetSetter(path []ottl.Field) (ottl.GetSetter[TransformContext], error) {
+	switch path[0].Name {
+	case "cache":
+		mapKey := path[0].Keys
+		if mapKey == nil {
+			return accessCache(), nil
+		}
+		return accessCacheKey(mapKey), nil
+	case "attributes":
+		mapKey := path[0].Keys
+		if mapKey == nil {
+			return accessSpanLinkAttributes(), nil
+		}
+		return accessSpanLinkAttributesKey(mapKey), nil
+	// TODO(fujin): Would it be valid to enable modifying the span id, is this the trace.link.span_id? It doesn't seem like there is something in 
+	// https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/ptrace/generated_spanlink.go that allows it to be set.
+	// case "span_id":
+	// 	return accessSpanLinkSpanID(), nil
```

Comment:
Yes, we should allow modifying it.  These functions should allow setting spanlink traceID and spanID: https://github.com/open-telemetry/opentelemetry-collector/blob/53615832e65416a89667611481b8765ab244021c/pdata/ptrace/generated_spanlink.go#L52-L72